nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -727,7 +727,8 @@ mod test {
                 }
             }
             for zone in &zones {
-                let OmicronZoneType::Nexus { .. } = zone.zone_type else {
+                let OmicronZoneType::Nexus { .. } = zone.config.zone_type
+                else {
                     panic!("unexpectedly added a non-Crucible zone: {zone:?}");
```

Comment:
Same suggestion as above about using `zone_type.is_nexus()`, and can we fix the comment string? (Both of these were problems prior to the PR, almost certainly from me - sorry!)

Reply:
Fixed.